src/org/labkey/snd/SNDManager.java

```diff
@@ -2300,14 +2300,13 @@ private String insertExpObjectProperties(Container c, User u, Event event, Event
 
         for (AttributeData attributeData : eventData.getAttributes())
         {
-            if (attributeData.getPropertyName() != null)
-            {
-                propertyDescriptor = OntologyManager.getPropertyDescriptor(PackageDomainKind.getDomainURI(
-                        SNDSchema.NAME, PackageDomainKind.getPackageKindName(), c, u) + "-" + pkgId + "#" + attributeData.getPropertyName(), c);
-            }
-            else
+            propertyDescriptor = OntologyManager.getPropertyDescriptor(attributeData.getPropertyId());
+
+            if (propertyDescriptor == null)
             {
-                propertyDescriptor = OntologyManager.getPropertyDescriptor(attributeData.getPropertyId());
+               propertyDescriptor = OntologyManager.getPropertyDescriptor(PackageDomainKind.getDomainURI(
+                        SNDSchema.NAME, PackageDomainKind.getPackageKindName(), c, u) + "-" + pkgId + "#" +
+                        attributeData.getPropertyName().replaceAll("\\s", "+"), c);
```

Comment:
`PageFlowUtil.encode(attributeData.getPropertyName())`

Might need to do a quick test on this, but this should handle how these names are expected to be encoded.

Reply:
Hi Marty,

Thanks for looking at the PR. Unfortunately, your suggestion is not going to work. The PageFlowUtil.encode() method replaces the space with a %20, resulting in, for example, body%20part. In the attribute data, the propertyURI replaces the space with a +. Here is the propertyURI for the body part attribute:

urn:lsid:txbiomed.org:package-snd.Folder-6:Package-62#body+part

Both the + and the %20 are proper URL encodings for the space, but the OntologyManager.getPropertyDescriptor() function is doing a key/value lookup in the property descriptor cache, with the propertyURI as the key, and it is failing when the encoding doesn’t match.

Reply:
Sorry I didn't do enough investigation into that. Here is the function that is being used when those are first encoded. Let me know if that works for you.

`Lsid.encodePart(attributeData.getPropertyName())`

Reply:
That does the trick! I'll submit a new PR for the change. Thanks!